doc/dev/mgmt/msrest_to_dpg_migration.md

```diff
@@ -0,0 +1,247 @@
+# Azure SDK Migration Guide: Swagger to TypeSpec Generation Breaking Changes
+
+This guide covers the breaking changes you'll encounter when upgrading from Swagger-generated SDKs to our new TypeSpec-generated SDKs and how to fix them in your code.
+Summary of Breaking Changes
+
+When migrating from Swagger to TypeSpec-generated SDKs, expect these breaking changes:
+
+| Change                                                                              | Impact                                                    | Quick Fix                                                                         |
+| ----------------------------------------------------------------------------------- | --------------------------------------------------------- | --------------------------------------------------------------------------------- |
+| [Dictionary Access](#dictionary-access-syntax)                                      | `as_dict()` parameter renamed, output format changed      | Recommended removal of `as_dict()` and directly access model, or replace `keep_readonly=True` with `exclude_readonly=False`, expect `camelCase` keys |
+| [Model Hierarchy](#model-hierarchy-reflects-rest-api-structure)                     | Multi-level flattened properties removed                  | Replace `obj.level1_level2_prop` with `obj.level1.level2.prop`                    |
+| [Additional Properties](#additional-properties-handling)                            | `additional_properties` parameter removed                 | Use direct dictionary syntax: `model["key"] = value`                              |
+| [String Representation](#string-representation-matches-rest-api)                    | Model key output changed from `snake_case` to `camelCase` | Update any code parsing model strings to expect `camelCase`                       |
+| [Serialization/Deserialization](#serialization-and-deserialization-methods-removed) | Manual serialization methods removed                      | Use dictionary access for serialization, constructor for deserialization          |
+
+## Detailed Breaking Changes
+
+### Dictionary Access Syntax
+
+**What changed**: TypeSpec-generated models support direct dictionary access and use different parameter names and output formats compared to Swagger-generated models.
```

Comment:
Can't talk about Swagger or TypeSpec in a public document. Last time I put that comment, it applies for the whole PR :)